sympy/polys/polytools.py

```diff
@@ -5307,6 +5307,105 @@ def half_gcdex(f, g, *gens, **args):
         return s, h
 
 
+@public
+def extended_euclidean_algorithm(f, g):
```

Comment:
In SymPy the extended Euclidean algorithm is referred to as `gcdex`. There are a few functions with this name including one in this file.

Probably this function should be called something like `igcdex` since it returns an iterator over the polynomial remainder sequence.

Reply:
Or perhaps `gcdex_steps` or something.